EIPS/eip-draft_nft_updatable.md

```diff
@@ -0,0 +1,233 @@
+
+The on-chain log trace of updates guarantee that anyone can recalculate and verify independently the current updated metadata starting from the original metadata. The fact that the calculation is deterministic allows easy caching of intermediate transformations and the efficient processing of new updates using these caches.
+
+The number of updates defined by each event is to be determined by the smart contract logic and use case, but it can easily scale to thousands or millions of updates per event. The function(s) that should emit `MetadataUpdates` and the frequency of these on-chain updates is left at the discretion of this standard implementation.
+
+## Reference Implementation
+
+### Transformation engines
+
+We have been experimenting with this generic Metadata update proposal using the JSONata transformation language. 
+
+Here is a very simple example of a NFT metadata for an imaginary 'little monster' game :
+
+```json
+{
+    "name": "Monster 1",
+    "description": "Little monsters you can play with.",
+    "attributes": [
+      { "trait_type": "Level", "value": 0 },
+      { "trait_type": "Stamina", "value": 100 },
+    ],
+    "updatable": {
+      "engine": "jsonata@1.8.*",
+      "recipes": {
+        "levelUp": {
+          "eval": "$ ~> | attributes[trait_type='Level'] | {'value': value + 1} |"
+        },
+        "updateDescription": {
+          "eval": "$ ~> | $ | {'description': $newDescription} |"
+        }
+      }
+    }
+}
+ ```
+
+This updatable metadata can only be updated to increment by one the trait attribute "Level".
+
+An example JSON updates metadata would be :
+```json
+{
+    "updates": [
+      {"tokenId":"1","action":"levelUp"},
+      {"tokenId":"2","action":"levelUp"},
+      {"tokenId":"1","action":"updateDescription","args":{"newDescription":"Now I'm a big monster"}},
+      {"tokenId":"1","action":"levelUp"},
+      {"tokenId":"3","action":"levelUp"}
+    ]
+}
+ ```
+
+## Caveats
+
+Smart contracts should be careful and conscious of using this extension and still allow the metadata URI to be updated in some contexts (by not having the same URI returned by `tokenURI` or `uri` for a given tokenId). They need to take into account if previous changes could have been already broadcasted for that NFT, if these changes are compatible with the new "original metadata" and what semantic they decide to associate by combining these two kinds of "updates".
+
+## Backwards Compatibility
+
+The proposal is fully compatible with both ERC-721 and ERC-1155. Third party applications that don't support Updatable Metadata Extension will still be able to use the original metadata for each NFT.
+
+## Gas efficiency
+
+The proposal is extremely gas efficient, since gas costs are only proportional to the frequency of committing changes. Many changes for many tokens can be batched in one transaction for the cost of only one `emit`.
+
```

Comment:
This is not a valid EIP section.  Please move this content as a subsection of a section defined in https://raw.githubusercontent.com/ethereum/EIPs/master/eip-template.md